shell/platform/common/cpp/geometry.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_GEOMETRY_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_GEOMETRY_H_
+
+#include <cmath>
+
+namespace flutter {
+
+// A rectangle with position in Cartesian space specified relative to a
+// separately-maintained origin.
+class Rect {
+ public:
+  Rect() = default;
+  Rect(double left, double top, double right, double bottom)
+    : left_(left), top_(top), right_(right), bottom_(bottom) {}
+  Rect(const Rect& rect) = default;
+  Rect& operator=(const Rect& other) = default;
+
+  double left() const { return left_; }
+  double top() const { return top_; }
+  double right() const { return right_; }
+  double bottom() const { return bottom_; }
+  double width() const { return std::abs(right_ - left_); }
+  double height() const { return std::abs(top_ - bottom_); }
+
+ private:
+  double left_ = 0.0;
+  double top_ = 0.0;
+  double right_ = 0.0;
+  double bottom_ = 0.0;
+};
```

Comment:
Unfortunately LTRB is how we do it in the framework, so doing the same here for consistency. Amusingly, the message we get from the framework uses x,y + width,height.

Reply:
:/

I'd rather use origin/size than match the framework, personally, but we can try this and see how well it fits our use cases.

Reply:
Yeah between the messages over the channel and the transform, it's pretty ugly. SemanticsNodes coming to the a11y bridge on the other hand use LTRB rects, which is why I ended up going this way. I could always switch it back to offset/size and deal with a11y when we get to it.

Reply:
After further discussion with a11y folks, they're implementing in terms of point size. I'll go with that to make it easier to consolidate this work with theirs.